syntax = "proto3";
package msgs;

enum MessageType {
    AddOrder = 0;
    CancelOrder = 1;
    ClearBooks = 2;
}

message BaseMessage {
    MessageType type = 1;
    bytes payload = 2;
}

message AddOrderMsg {
    string symbol = 1;
    int32 userId = 2;
    int32 orderId = 3;
    int32 price = 4;
    int32 quantity = 5;
    bool isBuy = 6;
}

message CancelOrderMsg {
    int32 userId = 1;
    int32 orderId = 2;
}

message ClearBooksMsg {
}
